File: app/src/main/java/com/textfield/json/outabus/util/SortByDistance.java
package com.textfield.json.outabus.util;

import android.location.Location;

import com.textfield.json.outabus.Stop;

import java.util.Comparator;

/**
 * Created by Jason on 06/10/2015.
 */
public class SortByDistance implements Comparator<Stop> {
    double lat, lng;

    public SortByDistance(double lat, double lng) {
        this.lat = lat;
        this.lng = lng;
    }

    @Override
    public int compare(Stop lhs, Stop rhs) {
        float[] distance1 = new float[1];
        float[] distance2 = new float[1];
        Location.distanceBetween(lat, lng, lhs.getLat(), lhs.getLng(), distance1);
        Location.distanceBetween(lat, lng, rhs.getLat(), rhs.getLng(), distance2);
        if (distance1[0] > distance2[0]) return 1;
        if (distance1[0] < distance2[0]) return -1;
        return 0;
    }
}

File: app/src/main/java/com/textfield/json/outabus/util/SortByDistance.kt
package com.textfield.json.outabus.util

import android.location.Location

import com.textfield.json.outabus.entities.Stop

import java.util.Comparator

/**
 * Created by Jason on 06/10/2015.
 */
class SortByDistance(private val userLocation: Location) : Comparator<Stop> {
    override fun compare(lhs: Stop, rhs: Stop): Int {
        val distance1 = FloatArray(1)
        val distance2 = FloatArray(1)
        Location.distanceBetween(userLocation.latitude, userLocation.longitude, lhs.lat, lhs.lng, distance1)
        Location.distanceBetween(userLocation.latitude, userLocation.longitude, rhs.lat, rhs.lng, distance2)
        return distance1[0].compareTo(distance2[0])
    }
}

